flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -396,13 +415,11 @@ public int getBuffersInBacklog() {
 
 	private boolean shouldNotifyDataAvailable() {
 		// Notify only when we added first finished buffer.
-		return readView != null && !flushRequested && getNumberOfFinishedBuffers() == 1;
+		return readView != null && !flushRequested && !isBlockedByCheckpoint && getNumberOfFinishedBuffers() == 1;
 	}
 
 	private void notifyDataAvailable() {
-		if (readView != null) {
```

Comment:
better to make this change as a separate hotfix commit

Reply:
I remove the ```readView != null``` condition in check ```flush``` as suggested, so I add this condition back.

Reply:
Actually this condition even this method can be removed if we can reuse the preceding three conditions inside `shouldNotifyDataAvailable` for `#flush()` operation, because only the available conditions inside `#flush` not check the `readerView != null`.